Flow: Problem with exports using Git Module Loader in Helm

### What's wrong?

In order to test out a simple module with the Flow agent, I have created the following module:
```
discovery.kubernetes "pods" {
  role = "pod"
}
```

Then I load the module like this

```
module.git "module_discovery" {
    repository = "https://github.com/PUBLIC_REPO.git"
    revision   = "main"
    path       = "pod_discovery.river"
}
```

So far, so good. The module is loaded, and I can see it show up as healthy in the Graph UI.

![image](https://github.com/grafana/agent/assets/11229444/2ae5e8fa-f7a8-4252-9c2b-0253227ad8a3)

The first problem I encounter is that I get an error when I click on the View button in the discovery component in the "Module components" section
![image](https://github.com/grafana/agent/assets/11229444/f0cefae2-6859-4bc2-a181-e024ff974957)
![image](https://github.com/grafana/agent/assets/11229444/7f5a756e-4755-453c-8050-66c01d4e0eea)


Next, I continue to add an export to my module
```
export "simple" {
  value = discovery.kubernetes.pods.targets
}
```

For a short while, I can see the export appearing in the Graph UI
![image](https://github.com/grafana/agent/assets/11229444/834ecd0f-472e-4b39-b294-c5be0c17ec47)

However, after 20-30 seconds, the grafana-agent pods go into CrashLoopBackOff. If I inspect the logs of the pod, I can see the following error:
```
panic: runtime error: comparing uncomparable type []interface {}

goroutine 177 [running]:
github.com/grafana/agent/pkg/flow/internal/controller.(*valueCache).CacheModuleExportValue(0xc000943540, {0xc001190e71, 0x6}, {0x56423e0?, 0xc000013f08})
	/src/agent/pkg/flow/internal/controller/value_cache.go:76 +0xc8
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).EvaluateDependencies.func2({0x7954240?, 0xc000c58a00})
	/src/agent/pkg/flow/internal/controller/loader.go:443 +0x436
github.com/grafana/agent/pkg/flow/internal/dag.WalkReverse(0xc001284580, {0xc00163fba8, 0x1, 0x3?}, 0xc001647d80)
	/src/agent/pkg/flow/internal/dag/walk.go:69 +0x1d9
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).EvaluateDependencies(0xc0012b07e0, 0x0, 0xc000ec6000)
	/src/agent/pkg/flow/internal/controller/loader.go:424 +0x84e
github.com/grafana/agent/pkg/flow.(*Flow).Run(0xc000cd7380, {0x79abb70, 0xc000c6c8c0})
	/src/agent/pkg/flow/flow.go:213 +0x5af
github.com/grafana/agent/component/module.(*ModuleComponent).RunFlowController(0x709dea?, {0x79abb70?, 0xc000c6c8c0?})
	/src/agent/component/module/module.go:98 +0x27
created by github.com/grafana/agent/component/module/git.(*Component).Run
	/src/agent/component/module/git/git.go:105 +0x114
```

### Steps to reproduce

All steps are provided in the description

### System information

Azure Kubernetes 1.24.9

### Software version

Grafana Agent Helm Chart v0.14.0

### Configuration

```text
# Module
discovery.kubernetes "pods" {
  role = "pod"
}

export "simple" {
  value = discovery.kubernetes.pods.targets
}

# Values.yml
agent:
  enableReporting: false
  configMap:
    create: true
    content: '

      module.git "module_discovery" {
        repository = "https://github.com/PUBLIC_REPO.git"

        revision   = "main"

        path       = "pod_discovery.river"
      }

    '
```


### Logs

```text
ts=2023-05-27T17:58:26.089451397Z level=info trace_id=3a5d69c9bba2a4c1a445c181f49416d7 msg="starting complete graph evaluation"
ts=2023-05-27T17:58:26.091076565Z level=info msg="now listening for http traffic" addr=0.0.0.0:80
ts=2023-05-27T17:58:26.718062954Z component=module.git.module_discovery level=info trace_id=a4bcdf28b167c6a5c3543334a22918de msg="starting complete graph evaluation"
ts=2023-05-27T17:58:26.718370448Z component=module.git.module_discovery/discovery.kubernetes.pods level=info msg="Using pod service account via in-cluster config"
ts=2023-05-27T17:58:26.718895038Z component=module.git.module_discovery level=info trace_id=a4bcdf28b167c6a5c3543334a22918de msg="finished node evaluation" node_id=discovery.kubernetes.pods duration=770.685µs
ts=2023-05-27T17:58:26.718977536Z component=module.git.module_discovery level=info trace_id=a4bcdf28b167c6a5c3543334a22918de msg="finished node evaluation" node_id=export.simple duration=38.099µs
ts=2023-05-27T17:58:26.719006836Z component=module.git.module_discovery level=info trace_id=a4bcdf28b167c6a5c3543334a22918de msg="finished complete graph evaluation" duration=1.000281ms
ts=2023-05-27T17:58:26.719032735Z level=info trace_id=3a5d69c9bba2a4c1a445c181f49416d7 msg="finished node evaluation" node_id=module.git.module_discovery duration=629.509639ms
ts=2023-05-27T17:58:26.719058235Z level=info trace_id=3a5d69c9bba2a4c1a445c181f49416d7 msg="finished node evaluation" node_id=tracing duration=12.5µs
ts=2023-05-27T17:58:26.719088034Z level=info trace_id=3a5d69c9bba2a4c1a445c181f49416d7 msg="finished node evaluation" node_id=logging duration=9.6µs
ts=2023-05-27T17:58:26.719098834Z level=info trace_id=3a5d69c9bba2a4c1a445c181f49416d7 msg="finished complete graph evaluation" duration=629.717535ms
ts=2023-05-27T17:58:26.719162133Z level=info msg="scheduling loaded components"
ts=2023-05-27T17:58:26.719218531Z component=module.git.module_discovery level=info msg="updating repository pull frequency" new_frequency=1m0s
ts=2023-05-27T17:58:26.719236231Z component=module.git.module_discovery level=info msg="scheduling loaded components"
ts=2023-05-27T17:58:31.729122518Z component=module.git.module_discovery level=info trace_id=a260f620d2eacfcc571e95e5974b2670 msg="starting partial graph evaluation"
ts=2023-05-27T17:58:31.735688333Z component=module.git.module_discovery level=info trace_id=a260f620d2eacfcc571e95e5974b2670 msg="finished partial graph evaluation" duration=6.590115ms
ts=2023-05-27T17:58:31.735803533Z component=module.git.module_discovery/discovery.kubernetes.pods level=info msg="component exited"
panic: runtime error: comparing uncomparable type []interface {}

goroutine 177 [running]:
github.com/grafana/agent/pkg/flow/internal/controller.(*valueCache).CacheModuleExportValue(0xc000943540, {0xc001190e71, 0x6}, {0x56423e0?, 0xc000013f08})
	/src/agent/pkg/flow/internal/controller/value_cache.go:76 +0xc8
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).EvaluateDependencies.func2({0x7954240?, 0xc000c58a00})
	/src/agent/pkg/flow/internal/controller/loader.go:443 +0x436
github.com/grafana/agent/pkg/flow/internal/dag.WalkReverse(0xc001284580, {0xc00163fba8, 0x1, 0x3?}, 0xc001647d80)
	/src/agent/pkg/flow/internal/dag/walk.go:69 +0x1d9
github.com/grafana/agent/pkg/flow/internal/controller.(*Loader).EvaluateDependencies(0xc0012b07e0, 0x0, 0xc000ec6000)
	/src/agent/pkg/flow/internal/controller/loader.go:424 +0x84e
github.com/grafana/agent/pkg/flow.(*Flow).Run(0xc000cd7380, {0x79abb70, 0xc000c6c8c0})
	/src/agent/pkg/flow/flow.go:213 +0x5af
github.com/grafana/agent/component/module.(*ModuleComponent).RunFlowController(0x709dea?, {0x79abb70?, 0xc000c6c8c0?})
	/src/agent/component/module/module.go:98 +0x27
created by github.com/grafana/agent/component/module/git.(*Component).Run
	/src/agent/component/module/git/git.go:105 +0x114
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,7 +19,7 @@ Main (unreleased)
 - Upgrade the embedded windows_exporter to commit 79781c6. (@jkroepke)
 
 - Prometheus exporters in Flow mode now set the `instance` label to a value similar to the one they used to have in Static mode (<hostname> by default, customized by some integrations). (@jcreixell)
- 
+
 - `phlare.scrape` and `phlare.write` have been renamed to `pyroscope.scrape` and `pyroscope.scrape`. (@korniltsev)
 
 ### Features
@@ -104,6 +104,8 @@ Main (unreleased)
 - Fix panic in `prometheus.operator.servicemonitors` from relabel rules without certain defaults. (@captncraig)
 
 - Fix issue in modules export cache throwing uncomparable errors. (@mattdurham)
+
+- Fix issue where the UI could not navigate to components loaded by modules. (@rfratto)
 
 ### Other changes
 

--- a/web/ui/src/features/component/ComponentList.tsx
+++ b/web/ui/src/features/component/ComponentList.tsx
@@ -1,4 +1,3 @@
-import React from 'react';
 import { NavLink } from 'react-router-dom';
 
 import { HealthLabel } from '../component/HealthLabel';
@@ -15,8 +14,9 @@ interface ComponentListProps {
 
 const TABLEHEADERS = ['Health', 'ID'];
 
-const ComponentList = ({ components }: ComponentListProps) => {
+const ComponentList = ({ components, parent }: ComponentListProps) => {
   const tableStyles = { width: '130px' };
+  const pathPrefix = parent ? parent + '/' : '';
 
   /**
    * Custom renderer for table data
@@ -29,7 +29,7 @@ const ComponentList = ({ components }: ComponentListProps) => {
         </td>
         <td className={styles.idColumn}>
           <span className={styles.idName}>{id}</span>
-          <NavLink to={'/component/' + id} className={styles.viewButton}>
+          <NavLink to={'/component/' + pathPrefix + id} className={styles.viewButton}>
             View
           </NavLink>
         </td>

